.github/workflows/pr-validation.yaml

```diff
@@ -0,0 +1,66 @@
+---
+# yamllint disable rule:line-length
+# Automatically runs on PRs to ensure the changes meet requirements and pass tests.
+name: PR Validation
+
+on:  # yamllint disable-line rule:truthy
+  pull_request:
+    branches: [master]
+    types: [opened, reopened, synchronize]
+  pull_request_target:
+    branches: [master]
+    types: [opened, reopened, synchronize]
+
+jobs:
+  pr-validation:
+    name: PR Validation
+    runs-on: ubuntu-latest
+    if: |
+      (github.actor != 'dependabot[bot]' && github.event_name != 'pull_request_target') ||
+      (github.actor == 'dependabot[bot]' && github.event_name == 'pull_request_target')
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+        if: github.actor != 'dependabot[bot]'
+
+      # See: https://hugo.alliau.me/2021/05/04/migration-to-github-native-dependabot-solutions-for-auto-merge-and-action-secrets/#share-your-secrets-with-dependabot
+      - name: Checkout Repository [Dependabot]
```

Comment:
Interesting. We could also override `GITHUB_TOKEN` with a RW one from secrets I think

Reply:
Shamelessly copied from https://github.com/RentTheRunway/dw_parent_pom/blob/f324deddb415496c01c81dbfe3c2151d2e1fee14/.github/workflows/pr-validation.yml#L39

One thing we probably could do with updating is the release jobs, to follow what we have for dw_parent_pom and such.

Hence my question about the open-source'ability of this. We might not be able to straight follow what we have for internal core libraries.